pkg/defaulting/configuration.go

```diff
@@ -292,18 +273,18 @@ var (
 			},
 		},
 		ProviderIncompatibilities: []kubermaticv1.Incompatibility{
-			// In-tree cloud providers have been fully removed in Kubernetes 1.29.
+			// In-tree cloud providers have been fully removed in Kubernetes 1.30.
 			// Thus, no in-tree provider is available anymore, and no cluster with in-tree CCM
 			// can be upgraded to 1.29.
 			{
 				Provider:  "",
-				Version:   ">= 1.29.0",
+				Version:   ">= 1.30.0",
 				Condition: kubermaticv1.InTreeCloudProviderCondition,
 				Operation: kubermaticv1.CreateOperation,
 			},
 			{
 				Provider:  "",
-				Version:   ">= 1.29.0",
+				Version:   ">= 1.30.0",
 				Condition: kubermaticv1.InTreeCloudProviderCondition,
 				Operation: kubermaticv1.UpdateOperation,
 			},
```

Comment:
Changing this to >=1.30 doesn't make sense, there can't be any 1.30 clusters with in-tree providers. I think we should keep this rule unchanged as it is for KKP 2.28, while we're still allowing the 1.29->1.30 upgrade path.